src/core/core.scale.js

```diff
@@ -593,8 +601,13 @@ var Scale = Element.extend({
 
 		me.handleMargins();
 
-		me.width = minSize.width;
-		me.height = minSize.height;
+		if (isHorizontal) {
```

Comment:
I'm not sure about dealing with orientation here since it's already done between lines 528 and 539. I would keep the previous code here and inject margins earlier to avoid duplicating the orientation logic. But also, now `me.width` doesn't depend anymore of `minSize` or `maxSize`, but now `chart.width` which I don't think is correct.

Reply:
The `width` is set to this value in `core.layouts`. The value that was being set here previously was discarded and had no effect since it was overriden by `core.layouts`. The `minSize` is unused by that code, which why I'd added a TODO to remove it in v3 since nothing actually uses it. By setting the final value here, we're able to do auto-skipping earlier (since we need to know the width of the scale to know how many ticks to fit)